subprojects/core/src/main/java/org/gradle/execution/taskgraph/DefaultTaskGraphExecuter.java

```diff
@@ -130,7 +133,13 @@ public void execute() {
             taskPlanExecutor.process(taskExecutionPlan, new EventFiringTaskWorker(taskExecuter.create(), buildOperationExecutor.getCurrentOperation()));
             LOGGER.debug("Timing: Executing the DAG took " + clock.getElapsed());
         } finally {
-            taskExecutionPlan.clear();
+            coordinationService.withStateLock(new Transformer<ResourceLockState.Disposition, ResourceLockState>() {
```

Comment:
Unless I'm missing something, I don't think it's necessary to surround this in a call to the coordination service.  At this point, the task plan executor should have finished processing the plan and stopped all of his workers, so you should have exclusive access to the plan.

Reply:
What happens if `taskPlanExecutor.process` throws an exception? In theory, In theory, there could be problems stopping the executor service, right?

If you still think it is safe to remove the `coordinatorService.withStateLock`, then I'll do so.

Reply:
Fair enough.  I think that's the something I was missing - I wasn't thinking about task plan executor blowing up and leaving threads running.  It's probably safer to clear out the plan this way, then.